src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+     *      {label: "AllWISE Source Catalog", value: "wise_allwise_p3as_psd", proj: "AllWISE Database", cat: Array[12]} and
+     *      Each 'cat' array ith element is the value representing the ith column name below
+     *      ["projectshort", "subtitle", "description", "server", "catname", "cols", "nrows", "coneradius", "infourl", "ddlink", "catSearchProcessor", "ddSearchProcessor"]
+     * @see master table file from http://irsa.ipac.caltech.edu/cgi-bin/Gator/nph-scan?mode=ascii
+     * @param {Object} master table model
+     * @returns {Void} set the state to include master table
+     */
+    setMaster(master = {}) {
+        this.setState(...this.state, {master});
+    }
+
+    render() {
+        const {master={}} = this.state;
+        if (isEmpty(master)) {
+            return (
+                <div style={{position: 'relative'}}>
+                    <div className='loading-mask'/>
+                </div>
+            );
+        }
+
+        return (
+            <FieldGroup groupKey={gkey}
+                        reducerFunc={userChangeDispatch()}
+                        keepState={true}>
+                <FieldGroupTabs initialState={{ value:'catalog' }} fieldKey='Tabs'>
+                    <Tab name='Search Catalogs' id='catalog'>
+                        <CatalogDDListConnected fieldKey='tableview' {...this.props} {...this.state} />
+                    </Tab>
+                    <Tab name='Load Catalog' id='loadcat'>
+                        <div
+                            style={{padding:5, display:'flex', flexDirection:'column', flexWrap:'no-wrap', alignItems:'center' }}>
+                            <FileUpload
+                                wrapperStyle={{margin: '5px 0'}}
+                                fieldKey='fileUpload'
+                                initialState={{
+                                            tooltip: 'Select an IPAC catalog table file to upload',
+                                            label: 'File:'}}
+                            />
+                            <div style={helpIdStyle}>
+                                <em style={{color:'gray'}}>Custom catalog in IPAC table format</em>
+                                <HelpIcon
+                                    helpid={'basics.loadcatalog'}/>
+                            </div>
+                        </div>
+                    </Tab>
+                    <Tab name='VO Catalog' id='vosearch'>
+                        <VoSearchPanel fieldKey='vopanel'/>
+                    </Tab>
+                </FieldGroupTabs>
+            </FieldGroup>
+
+        );
+    }
+}
+
+/**
+ * Reducer from field group component, should return updated project and sub-project updated
+ * @returns {Function} reducer to change fields when user interact with the dialog
+ */
+var userChangeDispatch = function () {
+
+    return (inFields, action) => {
+        if (!inFields) {
+            return fieldInit();
+        } else {
```

Comment:
This is picky and I never did it in the java days, but if you  do
```
if (!inFields) return fieldInit();
```
You don't need the else and it is less indention to have to manage.